FlagsmithClient/src/main/java/com/flagsmith/internal/FlagsmithEventService.kt

```diff
@@ -0,0 +1,89 @@
+package com.flagsmith.internal
+
+import android.util.Log
+import com.flagsmith.entities.FlagEvent
+import com.google.gson.Gson
+import kotlinx.coroutines.flow.MutableStateFlow
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.sse.EventSource
+import okhttp3.sse.EventSourceListener
+import okhttp3.sse.EventSources
+import java.util.concurrent.TimeUnit
+
+internal class FlagsmithEventService constructor(
+    private val eventSourceUrl: String?,
+    private val environmentKey: String,
+    private val updates: (Result<FlagEvent>) -> Unit
+) {
+    private val defaultEventSourceHost = "https://realtime.flagsmith.com/"
+
+    private val sseClient = OkHttpClient.Builder()
+        .addInterceptor(FlagsmithRetrofitService.envKeyInterceptor(environmentKey))
+        .connectTimeout(6, TimeUnit.SECONDS)
+        .readTimeout(10, TimeUnit.MINUTES)
+        .writeTimeout(10, TimeUnit.MINUTES)
+        .build()
+
+    private val defaultEventSourceUrl: String = defaultEventSourceHost + "sse/environments/" +  environmentKey + "/stream"
+
+    private val sseRequest = Request.Builder()
+        .url(eventSourceUrl ?: defaultEventSourceUrl)
```

Comment:
This seems different to the JS implementation where the user is able to define the host, but not the full URL. Is there a reason this logic is different? The only reason to expose this to the user is for self hosting purposes, the URI of the API endpoint will still be the same.

Reply:
This is sorted now, and more consistent with the main URL